### ISSUE TITLE
Delete doesn't invalidate the cache

### ISSUE THREAD

**Issue by nephlm** [Miserlou/NoDB#15]
If you instantiate NoDB and set cache to True, then you can save and object and delete the object.  

If you load the object at at this point, the deleted object will be returned since it's still in the cache.  Delete is not cache aware; it deletes from the s3 bucket, but doesn't update the cache.

```python
nodb=NoDB()
nodb.cache=True
nodb.index = 'name'
nodb.save({'name': 'jeff', 'age': 19})
nodb.delete('jeff')
nodb.load('jeff') #=> {'name': 'jeff', 'age': 19}
```
That's probably not the expected behavior.

**Comment by bendog** [comment]
should probably start by writing a test case for this

**Comment by bendog** [comment]
Merged into master, waiting on PyPi update
